src/client/Microsoft.Identity.Client/PlatformsCommon/DeviceAuthHelper.cs

```diff
@@ -16,7 +16,7 @@ internal class DeviceAuthHelper
         public static IDictionary<string, string> ParseChallengeData(HttpResponse response)
         {
             IDictionary<string, string> data = new Dictionary<string, string>();
-            string wwwAuthenticate = response.Headers.GetValues(PKeyAuthConstants.WwwAuthenticateHeader).FirstOrDefault();
+            string wwwAuthenticate = response.Headers.GetValues(PKeyAuthConstants.WwwAuthenticateHeader).Single();
```

Comment:
It's probably safer to return empty dictionary if header does not exist.

Reply:
The header will exist in this method. there is a method before that ensures the headers exist before parsing them.

Reply:
i agree, ill update